### PR TITLE
Add support for custom dns lookup

### PR DESCRIPTION
**Pull request by thomas510111**
Default dns.lookup() is running on libuv threadpool and on high load or issues with the dns server it can cause increased latency in getting the initial connection.
https://github.com/nodejs/node/issues/8436

In my opinion the best way to "patch" this is allow users to use their own dns lookup function.

### CODE REVIEW

**Comment by thomas510111** [comment]
By the way, I havent worked with typescript and couldnt figure out how to properly declare type of dnsLookup.

File: src/chain.ts [hunk 1]
```diff
@@ -22,11 +22,13 @@ interface Options {
     headers: string[];
     path?: string;
     localAddress?: string;
+    lookup?: any;
```
Comment (szmarczak) [inline src/chain.ts:25]: Please type this properly.

**Review by szmarczak — CHANGES_REQUESTED** [review]

File: src/chain.ts (hunk 1 above)
Comment (szmarczak) [reply src/chain.ts:25]: I think it should be possible to `dns['lookup']` or `typeof dns['lookup']`

**Comment by freeway7** [comment]
How to set custom DNS server?

**Pull request merged**